### PR TITLE
set end_comparator of LayerPruningModifier to -1

### PR DESCRIPTION
**Pull request by bfineran**
Under the current `end_comparator` the end epoch of `LayerPruningModifier` must be greater than the start epoch. This would make a schedule like `start_epoch=2.0, end_epoch=-1` invalid which is a bug, since we should be able to start at an arbitrary epoch (`start_epoch=2.0`) and run indefinitely (`end_epoch=-1.0`)

`end_comparator` specs: 
https://github.com/neuralmagic/sparseml/blob/29119f73596fbea01a0b82133696305995a5b72c/src/sparseml/optim/modifier.py#L630
https://github.com/neuralmagic/sparseml/blob/29119f73596fbea01a0b82133696305995a5b72c/src/sparseml/optim/modifier.py#L631

### CODE REVIEW

**Review by mgoin — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]

**Review by spacemanidol — APPROVED** [review]

File: src/sparseml/pytorch/sparsification/pruning/modifier_pruning_layer.py [hunk 1]
```diff
@@ -70,6 +70,7 @@ def __init__(
             start_epoch=start_epoch,
             end_epoch=end_epoch,
             update_frequency=-1.0,
+            end_comparator=-1,
```
Comment (bfineran) [inline src/sparseml/pytorch/sparsification/pruning/modifier_pruning_layer.py:73]: noting that we should not merge this PR yet. this fix uncovered potential issues with the logic surrounding the implementation of `end_comparator=-1`

**Comment by bfineran** [comment]
blocking landing - see unresolved comment

**Comment by dbogunowicz** [comment]
@bfineran what needs to be completed prior to this PR being merged?

**Review by markurtz — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Comment by bfineran** [comment]
@dbogunowicz see latest non merge commit - updated `end_comparator=-1` to only allow `end_epoch < start_epoch` if `end_epoch==-1`. a few more changes will be needed on the recipe composition side to be added in a future PR

**Pull request merged**